www/blak05/cv04/registration.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+$invalidInputs = [];
+
+function sendEmail($email, $subject, $message)
+ {
+
+     $header = [
+         'MIME-Version_ 1.0',
+         'Content-type: text/html, charset=utf-8',
+         'From: blak05@dev.com',
+         'Reply-To: app@dev.com',
+         'X-Mailer: PHP/8.0',
+     ];
+     $body = "
+         <h1>$subject</h1>
+         <p>$message</p>
+     ";
+     return mail($email, $subject, $body, $header);
+ }
+
+function makeRegistration($data){
+    $databaseFileName = __DIR__ . '/database/users.db';
+    $lines = file($databaseFileName);
+    $message = "";
+    $isExistingUser = false;
+    $success = false;
+    foreach($lines as $line){
+        if(!$line){
+            continue;
+        }
+        $fields = explode(';', $line);
+        $user = [
+            'name' => $fields[0],
+            'email' => $fields[1],
+            'password' => $fields[2],
+        ];
+
+        if($user['name'] === $data['name']){
+            $isExistingUser = true;
+            break;
+        }
+        if($user['email'] === $data['email']){
+            $isExistingUser = true;
+            break;
+        }
+
+    }
+
+    if($isExistingUser){
+        $message = "Uživatel již existuje";
+    }
+
+    if(!$isExistingUser){
+
+        $userInfo = [
+            $data['name'],
+            $data['email'],
+            $data['password'],
+        ];
+        //vyrobit záznam 
+        $newRecord = implode(';', $userInfo) . "\r\n";
+
+    //vložit do souboru
+        file_put_contents($databaseFileName, $newRecord, FILE_APPEND);
```

Comment:
mozna nemate nastavene pravo na zapis do souboru, nejde se mi prihlasit

Reply:
celé původní cv04 jsem smazal z esa a nahrál verzi, kterou jsem před minutou nahrál sem. A db opět nefungovala a bylo nutné ve filezille dát právo na zápis. Ale než jsem to uploadoval, tak jsem práva nastavil, nevím, jak bych to měl udělat, aby i při uploadu dával soubor práva pro zápis každému. Takže je možné, že chyba se u vás zase bude opakovat.